### ISSUE TITLE
System.Net.Sockets.SocketException (10060)

### ISSUE THREAD

**Issue by tthirmal** [sshnet/SSH.NET#684]
I am getting System.Net.Sockets.SocketException (10060), while downloading the files from SFTP into Azure Blob storage using C#, and this code is working fine in my local machine(running manually in Visual Studio 2019), but the problem is after published my code into WebJobs it's getting failed and giving attached Errors.

```
[05/23/2020 15:31:16 > 36bfb3: SYS INFO] Status changed to Initializing
[05/23/2020 15:31:16 > 36bfb3: SYS INFO] Job directory change detected: Job file 'FileDownloader.dll' timestamp differs between source and working directories.
[05/23/2020 15:31:16 > 36bfb3: SYS INFO] Run script 'run.cmd' with script host - 'WindowsScriptHost'
[05/23/2020 15:31:16 > 36bfb3: SYS INFO] Status changed to Running
[05/23/2020 15:31:17 > 36bfb3: INFO] 
[05/23/2020 15:31:17 > 36bfb3: INFO] D:\local\Temp\jobs\triggered\FileDownloader\v1dgfgmq.hla>dotnet FileDownloader.dll  
[05/23/2020 15:31:39 > 36bfb3: INFO] An exception has been caught System.Net.Sockets.SocketException (10060): A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
[05/23/2020 15:31:39 > 36bfb3: INFO]    at Renci.SshNet.Abstractions.SocketAbstraction.Connect(IPEndPoint remoteEndpoint, TimeSpan connectTimeout)
[05/23/2020 15:31:39 > 36bfb3: INFO]    at Renci.SshNet.Session.SocketConnect(String host, Int32 port)
[05/23/2020 15:31:39 > 36bfb3: INFO]    at Renci.SshNet.Session.Connect()
[05/23/2020 15:31:39 > 36bfb3: INFO]    at Renci.SshNet.BaseClient.Connect()
[05/23/2020 15:31:39 > 36bfb3: INFO]    at FileDownloader.WithSSHHostKey.DownloadFiles() in D:\<I am getting here local solution path even it's ran through webjob>\WithSSHHostKey.cs:line 29
[05/23/2020 15:31:39 > 36bfb3: SYS INFO] Status changed to Success

```

what is the issue here ?.

**Comment by sachetpatil1** [comment]
Did you got any solution??

I am also facing same issue. Although credentials works fine with same code in Dev and Test environment and files successfully transferring to server but after deploying in UAT it got failed with this error :(

**Comment by gavin1970** [comment]
Have you guys tried Using pscp From win client or scp from linux, to see if you get the same error?   Uses the same port is why I ask.

**Comment by sachetpatil1** [comment]
I have implemented this library in azure function for storing files into sftp server location. Code works fine in Development and test Function app but didnt worked after deploying in UAT function app.. I can transfer files through same libraries in lower environment but not sure what happn in UAT and Prod. If any issue in port then it basically should not work in dev and test also since we are connecting to same sftp server and credentials.

Currently stuck with this issue from last 2 days and now planning to replace it with winscp. :(

**Comment by darkoperator** [comment]
Are you using Core or NetStandard version of the library in Azure? I believe it would need to be one of those to work in UAT

**Comment by sachetpatil1** [comment]
We are using .net Core 2.1 and Azure function runtime version 2. Anything needs to change here?

**Comment by darinkes** [comment]
SocketException 10060 implies an connection problem on TCP/IP Layer.
The target cannot be reached, so there is no SSH communication at this point.
You should validate the host is actually reachable, e.g. check firewall, routing, ....

**Comment by dineshclinton** [comment]
I'm encountering the same issue. I could connect via the command line through SSH and SFTP but fails through Renci.SshNet in the same machine.

@sachetpatil1, Were you able to identify the root cause?

Thanks,
Dinesh

**Comment by sachetpatil1** [comment]
@dineshclinton After so much investigation, we decided to speak with our Azure admin team (different organisation) and got to know that they restricted FTP server to connect external resourves in firewall so that App service IP was not allowed to connect FTP server. We requested them to whitelist that IP and it worked.

In short, there is no issue in Renci dll regarding connection to external ftp. Now we are able to send all files but not sure how much file size it can handle (Curently we are sending max 100 mb file).

Let me know if this resolved your query and sorry for delayed reply :(
